packages/flutter/lib/src/material/slider_theme.dart

```diff
@@ -759,6 +772,41 @@ abstract class SliderTickMarkShape {
     bool isEnabled,
     TextDirection textDirection,
   });
+
+  /// Special instance of [SliderTickMarkShape] to skip the thumb painting.
+  ///
+  /// See also:
+  ///
+  /// * [SliderThemeData.tickMarkShape], which is the shape that the [Slider]
+  /// uses when painting tick marks.
+  static SliderTickMarkShape noTickMark = _EmptySliderTickMarkShape();
+}
+
+/// A special version of [SliderTickMarkShape] that has a zero size and paints
+/// nothing.
+///
+/// This class is used to create a special instance of a [SliderTickMarkShape]
+/// that will not paint any tick mark shape. A static reference is stored in
+/// [SliderTickMarkShape.noTickMark]. When this value  is specified for
+/// [SliderThemeData.tickMarkShape], the tick mark painting is skipped.
+class _EmptySliderTickMarkShape extends SliderTickMarkShape {
+  Size getPreferredSize({
```

Comment:
`getPreferredSize` and `paint` should probably be annotated with `@overrides` here.